src/main/scala/extractor/SemanticCtx.scala

```diff
@@ -162,27 +165,51 @@ abstract class SemanticDBWalker {
       Files.delete(file)
     }
   }
+  /**
+    * Function that, given two Results, it returns a Result that contains all the non-duplicate elements of both parameters.
+    *
+    * @param one
+    * @param other
+    * @return a Result that contains all the non-duplicate elements of one and other.
+    */
+  def mergeResults(one: Result, other: Result): Result = {
+    def mergeById[A <: ResultElement](one: Set[A], other: Set[A]): Set[A] = {
+      (one ++ other)
+        .groupBy(_.id)
+        .map{case (_, v) => v.head}
+        .toSet
+    }
+
+    Result(
+      mergeById(one.params, other.params),
+      one.funs ++ other.funs,
+      mergeById(one.links, other.links),
+      mergeById(one.implicits, other.implicits)
+    )
+  }
 }
 
-class SingleProjectWalker(rootPath: String) extends SemanticDBWalker {
+class SingleProjectWalker(rootPath: String) extends TreeWalker {
   val root = AbsolutePath(rootPath)
   println(s"Analyzing ${rootPath}")
-  def run[T](f: SemanticCtx => Iterable[(String, T)]): ConcurrentLinkedQueue[(String, T)] = {
+  def run[T](f: SemanticCtx => Result): Result = {
     import scala.collection.JavaConverters._
     deleteOldFiles(root)
-    val results = new ConcurrentLinkedQueue[(String, T)]
-    val files = Files
+    val results = Files
         .walk(root.toNIO)
         .iterator()
         .asScala
         .filter { file =>
           Files.isRegularFile(file) &&
             PathIO.extension(file) == "semanticdb"
         }
-        .toVector
+        .toSeq
         .par
-      files.foreach(x => SemanticDBFileVisitor(x, root, results, f))
-      println(" ")
-      results
+        .map {file => SemanticDBFileVisitor(file, f)}
+        .fold(Result(Set(), Seq(), Set(), Set())) {(acc, partial) =>
```

Comment:
I have seen this before: `Result(Set(), Seq(), Set(), Set())`

How about to create a constant in `Result` like:
```
object Result {
  val Empty = Result(Set(), Seq(), Set(), Set())
}
```

so you can write:

`fold(Result.Empty)(mergeResult)`